### PR TITLE
Merging master jdk8u282-b01 into openj9-staging

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Resolved merge conflict at
```
  jdk/make/Images.gmk
  make/common/NativeCompilation.gmk
```

Note: assuming https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/384/commits/faf2fd055c1ff916d837cb87f5fa387df8032b05 is superseded by the changeset from https://bugs.openjdk.java.net/browse/JDK-8252809

Tested in JDK8 `ppc64_aix`, `x86-64_linux`, `ppc64le_linux`, and `x86-64_windows`.

Signed-off-by: Jason Feng fengj@ca.ibm.com

### CODE REVIEW

File: make/common/NativeCompilation.gmk
Comment (keithc-ca) [inline make/common/NativeCompilation.gmk:452]: There are many unnecessary whitespace changes here relative to:
* 8025936: Windows .pdb and .map files does not have proper dependencies setup

File: make/common/NativeCompilation.gmk
Comment (keithc-ca) [inline make/common/NativeCompilation.gmk:494]: Relative to:
* 8025936: Windows .pdb and .map files does not have proper dependencies setup
this line is missing:
```
			$(TOUCH) $$@
```

**Review by keithc-ca — CHANGES_REQUESTED** [review]

**Comment by keithc-ca** [comment]
I think the goal for `NativeCompilation.gmk` should be [this](https://github.com/keithc-ca/openj9-openjdk8/blob/pr463/make/common/NativeCompilation.gmk).

File: make/common/NativeCompilation.gmk
Comment (JasonFengJ9) [reply make/common/NativeCompilation.gmk]: Fixed.

File: make/common/NativeCompilation.gmk
Comment (JasonFengJ9) [reply make/common/NativeCompilation.gmk]: Added it back.

**Comment by JasonFengJ9** [comment]
Thanks @keithc-ca , please have another look.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
Jenkins test sanity aix,osx,zlinux,win jdk8

**Comment by pshipton** [comment]
Since this is to the staging branch, in theory it could just be merged as it will go through an acceptance build afterwards.

**Comment by keithc-ca** [comment]
Before merging, I think this should be addressed:
```
14:46:15  cp: /Users/jenkins/workspace/Build_JDK8_x86-64_mac_Personal/build/macosx-x86_64-normal-server-release/jdk/objs/java_objs/java.dSYM is a directory (not copied).
14:46:15  make[2]: *** [/Users/jenkins/workspace/Build_JDK8_x86-64_mac_Personal/build/macosx-x86_64-normal-server-release/jdk/bin/java.dSYM] Error 1
```

**Comment by JasonFengJ9** [comment]
Looking at `Build_JDK8_x86-64_mac_Personal` failure (I was able to build it at local Mac machine).

**Comment by JasonFengJ9** [comment]
Updated the commit https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/463/commits/e61ba6df411cbf74e73cb871afce7e809ee3f1fa, and it worked in my personal MacOS build.

**Comment by JasonFengJ9** [comment]
Jenkins test sanity aix,osx,zlinux,win jdk8

**Comment by JasonFengJ9** [comment]
@keithc-ca could you re-start the PR build?

File: jdk/make/CompileLaunchers.gmk
Comment (keithc-ca) [inline jdk/make/CompileLaunchers.gmk:288]: This use of `-R` is inappropriate (the target is a file).

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: jdk/make/CompileLaunchers.gmk
Comment (JasonFengJ9) [reply jdk/make/CompileLaunchers.gmk:288]: It is to fix
```
20:01:43  rm: /Users/jenkins/workspace/Build_JDK8_x86-64_mac_Personal/build/macosx-x86_64-normal-server-release/jdk/bin/java.dSYM: is a directory
20:01:43  make[2]: *** [/Users/jenkins/workspace/Build_JDK8_x86-64_mac_Personal/build/macosx-x86_64-normal-server-release/jdk/bin/java.dSYM] Error 1
```
`java.dSYM` is a directory.

Comment (keithc-ca) [reply jdk/make/CompileLaunchers.gmk:288]: This rule describes how to make .../java.dSYM/Contents/Info.plist, which is a file.
We want to minimize the changes to openjdk code (the problem is not here anyway).

Comment (JasonFengJ9) [reply jdk/make/CompileLaunchers.gmk]: Removed this `-R`, and verified in a personal build. (local build works as before).

Also merged with latest `openj9` branch.

Please have another look.

**Comment by keithc-ca** [comment]
Jenkins compile aix,osx,zlinux,win jdk8

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
@pshipton can you please render an opinion on this?

**Review by pshipton — APPROVED** [review]

**Comment by pshipton** [comment]
Approved it. I'll let you do the merge. Merging will automatically kick off the acceptance build.

**Pull request merged**
